mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1138,18 +1145,19 @@ def _evaluate_custom_metrics(self, eval_df):
                 index=index,
                 name=custom_metric.name,
             )
-            metric_result = _evaluate_custom_metric(
+            metric_value = _evaluate_custom_metric(
                 custom_metric_tuple,
                 eval_df_copy,
                 copy.deepcopy(self.metrics),
                 copy.deepcopy(self.metrics_values),
             )
-            name = (
-                f"{custom_metric.name}/{custom_metric.version}"
-                if custom_metric.version
-                else custom_metric.name
-            )
-            self.metrics_values.update({name: metric_result})
+            if metric_value:
+                name = (
+                    f"{custom_metric.name}/{custom_metric.version}"
+                    if custom_metric.version
+                    else custom_metric.name
+                )
+                self.metrics_values.update({name: metric_value})
```

Comment:
I think we should still log it as None in the evaluation_results.json so there is no data loss or data discrepancy in the eval_results table.

Reply:
This is checking if the entire result (MetricValue) is None, it will still log if there are scores, justifications, or aggregate results that are None - do you think we should log None for every row if the function returns None?

Reply:
We will hit this case when something fails to load - e.x. import evaluate fails. It will log a warning and return None

Reply:
Got it. In that case, I agree. Lets ignore it :) Thanks